pig/src/main/java/org/jboss/pnc/bacon/pig/impl/utils/BuildFinderUtils.java

```diff
@@ -169,10 +170,14 @@ public static BuildConfig getKojiBuildFinderConfig() {
      * @return the builds, including not found files if includeNotFound is true, or the found builds otherwise
      */
     public static List<KojiBuild> findBuilds(File file, boolean includeNotFound) {
-        BuildConfig config = getKojiBuildFinderConfig();
         List<String> inputs = Collections.singletonList(file.getPath());
-        ExecutorService pool = Executors.newFixedThreadPool(2);
-        DistributionAnalyzer analyzer = new DistributionAnalyzer(inputs, config);
+        return findBuilds(includeNotFound, inputs, 2);
+    }
+
+    public static List<KojiBuild> findBuilds(boolean includeNotFound, Collection<String> inputs, int poolSize) {
+        BuildConfig config = getKojiBuildFinderConfig();
+        ExecutorService pool = Executors.newFixedThreadPool(poolSize);
```

Comment:
I'm pretty sure this isn't doing anything. Even when you set it to 50, you've still only got two tasks in there, not 50, because you're recreating the pool each time.

Reply:
oh it does a lot :) the inputs collection is quite big for Quarkus and it's visible that it's indeed using 50 threads for it.

Reply:
OK, I think I didn't follow what you are doing. Are you creating a new `DistributionAnalyzer` for every file? If the input is an archive, it should only need one, but I believe it sets the number of threads in the thread pool based on the number of CPUs.